### PR TITLE
chore: lint files that touch `team-provider-info.json`

### PR DESCRIPTION
**Pull request by edwardfoyle**
#### Description of changes

In preparation for the `team-provider-info` refactor, I've linted all the files that call `getTeamProviderInfo` or `setTeamProviderInfo`

#### Description of how you validated changes
no functional changes made. existing unit tests pass

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariablesHelper.ts [hunk 1]
```diff
@@ -301,26 +317,26 @@ const getStoredParameters = (resourceName: string): object => {
...
   const envName = currentEnvName || stateManager.getLocalEnvInfo().envName;
   const teamProviderInfo = stateManager.getTeamProviderInfo(projectPath, { throwIfNotExist: false });
   return _.get(teamProviderInfo, [envName, 'categories', categoryName, resourceName], {});
 };
 
-const setStoredKeyValue = (resourceName: string, newKeyValue: object): void => {
+const setStoredKeyValue = (resourceName: string, newKeyValue: $TSAny): void => {
   const projectPath = pathManager.findProjectRoot();
-  const envName = stateManager.getLocalEnvInfo().envName;
+  const { envName } = stateManager.getLocalEnvInfo();
```
Comment (jhockett) [inline packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariablesHelper.ts:339]:
```suggestion
  const { envName } = stateManager.getLocalEnvInfo(projectPath);
```

File: packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariablesHelper.ts [hunk 2]
```diff
@@ -260,20 +276,20 @@ const getStoredList = (resourceName: string): { cloudFormationParameterName: str
   const projectBackendDirPath = pathManager.getBackendDirPath();
   const resourcePath = path.join(projectBackendDirPath, categoryName, resourceName);
   const functionParameterFilePath = path.join(resourcePath, functionParametersFileName);
-  const functionParameters = (JSONUtilities.readJson(functionParameterFilePath, { throwIfNotExist: false }) as object) || {};
+  const functionParameters = (JSONUtilities.readJson(functionParameterFilePath, { throwIfNotExist: false }) as $TSAny) || {};
```
Comment (jhockett) [inline packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariablesHelper.ts:279]: nit: here and elsewhere in this file we can pass in type to readJson, and I think $TSObject can be used to be slightly stricter

```ts
  const functionParameters = JSONUtilities.readJson<$TSObject>(functionParameterFilePath, { throwIfNotExist: false }) || {};
```

File: packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerMigrationUtils.ts [hunk 3]
```diff
@@ -109,7 +129,7 @@ This change requires a migration. Amplify will create a new Lambda layer version
     layerConfiguration.permissions = [defaultLayerPermission];
   } else {
     layerConfiguration.permissions = [];
-    permissions.map(permission => {
+    permissions.forEach(permission => {
```
Comment (jhockett) [inline packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerMigrationUtils.ts:132]: *facepalm*, good catch!

File: packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerMigrationUtils.ts [hunk 4]
```diff
@@ -1,27 +1,44 @@
-import { $TSContext, exitOnNextTick, JSONUtilities, pathManager, stateManager } from 'amplify-cli-core';
+import {
+  $TSContext, exitOnNextTick, JSONUtilities, pathManager, stateManager,
+} from 'amplify-cli-core';
+import { printer } from 'amplify-prompts';
```
Comment (jhockett) [inline packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerMigrationUtils.ts:4]: nit: line 128 can use printer as well (GH won't let me comment on the line)

File: packages/amplify-cli-core/src/permissionsBoundaryState.ts [hunk 5]
```diff
@@ -1,14 +1,20 @@
-import { stateManager } from './state-manager';
 import _ from 'lodash';
-import { $TSObject } from '.';
+// eslint-disable-next-line import/no-cycle
+import { stateManager } from './state-manager';
+// eslint-disable-next-line import/no-cycle
+import { $TSAny, $TSObject } from '.';
 
-let preInitTeamProviderInfo: any;
+let preInitTeamProviderInfo: $TSAny;
```
Comment (jhockett) [inline packages/amplify-cli-core/src/permissionsBoundaryState.ts:7]: nit: I think it's ideal to use `$TSTeamProviderInfo`, but it's just an alias for `$TSAny` currently.

**Review by danielleadams — APPROVED** [review]

**Comment by danielleadams** [comment]
Looks like there are 2 LGTM warnings that would be good to fix in this too ^^

File: yarn.lock [hunk 6]
```diff
@@ -9194,6 +9194,40 @@ amdefine@>=0.0.4:
   resolved "https://registry.npmjs.org/amdefine/-/amdefine-1.0.1.tgz#4a5282ac164729e93619bcfd3ad151f817ce91f5"
   integrity sha1-SlKCrBZHKek2Gbz9OtFR+BfOkfU=
 
+amplify-cli-core@2.6.0:
```
Comment (jhockett) [inline yarn.lock:9197]: These changes are unrelated to this PR - should be fixed by https://github.com/aws-amplify/amplify-cli/pull/10392

**Review by jhockett — APPROVED** [review]
Left a handful of nits, but LGTM

File: packages/amplify-cli-core/src/permissionsBoundaryState.ts (hunk 5 above)
Comment (edwardfoyle) [reply packages/amplify-cli-core/src/permissionsBoundaryState.ts:7]: with any luck, team provider info won't exist in a few months ;) but I'll update this for now

**Review by johnpc — APPROVED** [review]
